### DISCUSSION TITLE
Clarifications on the "parent-data.csv" dataset

### DISCUSSION THREAD

**Discussion by dufanrong** [facebookresearch/distributed_traces#6] [Q&A] [answered]
Hello,

I have been exploring your dataset **parent-data.csv** recently and I have a few questions regarding the interpretation of some data points. I hope you can provide some clarifications to help me better understand the dataset. Here are my questions:

1.In the `children_set` field, does the order of services represent the order in which the parent called them? For example, in the provided data, there are two rows with the same `c_set_index`, but the order of services in the `children_set` is different. Does this signify that the parent with ID `12030-00001` called the services within `c_set_index` 6 in a different order in these two requests?
| parent_name|children_set|c_set_index|
|  ----  | ----  | ----  | 
| 12030-00001|{'AADB', 'AAAF', 'AAAD', 'AAAB', 'AAAE', 'AABJ', 'AACA', 'AADF', 'AAAI'}| 6|
| 12030-00001|{'AADB', 'AAAF', 'AAAD', 'AAAE', 'AACA', 'AABJ', 'AAAB', 'AADF', 'AAAI'}| 6|

2.Does`num_calls` represent the total number of times the parent calls the `children_set` in a single request?

3.Is `num_calls` the count of calls made to each individual service within the entire `children_set` or the cumulative total count for all services? For instance, if a `children_set` contains four services (a, b, c, d), and each of them is called once in a single parent request, is the `num_calls` value 1 or 4?

4.Within a `children_set`, do all services have an equal number of calls, or is there a scenario where some services are called frequently while others are called infrequently?

Your insights on these questions will greatly assist me in using and interpreting the dataset effectively. I appreciate your time and assistance.

Best regards

**Comment by yurishkuro** [comment, marked as answer]
1. There is no meaning to the order, it's an unordered set. A "child" is a `service + endpoint` combination. A single invocation of the parent can make multiple calls to the same child, the set only keeps track of unique children IDs, not the count of calls, nor the order of calls.

2.  `num_calls` represents the total number of outbound calls from the parent. It cannot be less than `size(children_set)`, but can be larger if some children are called more than once.

3. `num_calls` is a total count: `SUM(numOfCallsToChild(x)) for each x in children_set`

4. Different children in `children_set` can have different number of calls. This number is determined by the business logic of the parent service, so highly unlikely to be equal across children when `num_calls >> size(children_set)`.

**Comment by dufanrong** [comment]
Thank you for your prompt response and for providing clarification on my previous questions. Your clarifications were extremely valuable, and I now have a much better understanding of the dataset. I appreciate your time and support！
